crates/polars-time/src/chunkedarray/rolling_window/dispatch.rs

```diff
@@ -98,15 +99,18 @@ where
             dt,
             "date/datetime"),
     };
+    let ca = ca.rechunk();
+    let by = by.rechunk();
+    let by_is_sorted = by.is_sorted(SortOptions {
+        descending: false,
+        ..Default::default()
+    })?;
     let by = by.datetime().unwrap();
     let tu = by.time_unit();
 
     let func = rolling_agg_fn_dynamic;
-    let out: ArrayRef = if matches!(by.is_sorted_flag(), IsSorted::Ascending) {
+    let out: ArrayRef = if by_is_sorted {
```

Comment:
This checks the sortedness flags and otherwise does a fast sortedness check that fails fast on non-sorted arrays.